app/Allocator/Squawk/General/AirfieldPairingSquawkAllocator.php

```diff
@@ -23,41 +23,60 @@ private function getPossibleRangesForFlight(string $origin, string $destination)
     public function allocate(string $callsign, array $details): ?SquawkAssignmentInterface
     {
         if (!isset($details['origin'])) {
-            Log::error('Origin not provided for airfield pairing squawk allocation', [$callsign, $details]);
+            Log::error(
+                'Origin not provided for airfield pairing squawk allocation',
+                [$callsign, $details]
+            );
             return null;
         }
 
         if (!isset($details['destination'])) {
-            Log::error('Destination not provided for airfield pairing squawk allocation', [$callsign, $details]);
+            Log::error(
+                'Destination not provided for airfield pairing squawk allocation',
+                [$callsign, $details]
+            );
             return null;
         }
 
         $assignment = null;
-        $this->getPossibleRangesForFlight($details['origin'], $details['destination'])->each(
-            function (AirfieldPairingSquawkRange $range) use (&$assignment, $callsign) {
-                $allSquawks = $range->getAllSquawksInRange();
-                $possibleSquawks = $allSquawks->diff(
-                    AirfieldPairingSquawkAssignment::whereIn('code', $allSquawks)->pluck('code')->all()
-                );
+        AirfieldPairingSquawkRange::getConnectionResolver()->connection()->transaction(
```

Comment:
Out of curiosity, is there a reason this syntax is used instead of a generic DB transaction? i.e. like https://laravel.com/docs/7.x/database#database-transactions

If they do the same thing, it might make it a bit more readable? Could even use `DB::beginTransaction();` to eliminate the need for the closure and passing in the required variables

Reply:
Changed!